src/instruction.cpp

```diff
@@ -82,7 +95,15 @@ bool instruction::valid() const
     {
         try
         {
-            computed = compute_shape(op, arguments);
+            if(module_args.empty())
+            {
+                computed = compute_shape(op, arguments);
+            }
+            else
+            {
+                auto out_shapes = compute_shape(module_args[0]);
```

Comment:
The output shape won't always be output shape of the module. I think the operator class will need a compute_shape that takes arguments and modules as inputs(`compute_shape(op, arguments, module_args)`).